### ISSUE TITLE
Var tog min kod vägen?

### ISSUE THREAD

**Issue by Masssssy** [kodsnack/advent_of_code_2017#233]
Min submodul fanns med igår och i morse men nu är den puts väck :( Vad har hänt där tro?

**Comment by bjoreman** [comment]
Antar att någon lyckades pusha en senare uppdaterad version av submodules som inte innehöll din modul och dessutom inte sågs som en konflikt? Submodules känns lite osäkra i ett repo med såpass många bidragsgivare.

**Comment by olssonm** [comment]
Verkar ha skett ett märkligt misstag där; ser ut som @mrinlumino råkade radera din submodul i commit  https://github.com/kodsnack/advent_of_code_2017/pull/218/commits/afab6642244ec890e286dcca42592bd2f38243f2 (`Submodule masssssy-python deleted from 83b2e4`) och ändringen merge:ades sedan in i PR #218.

Nog ganska lätt hänt tyvärr =/ Men nu är du tillbaka i alla fall =)

**Comment by bjoreman** [comment]
Ja, det är ju skönt att det går hyfsat lätt att reparera skadan. Men störigt ändå.

**Comment by bjoreman** [comment]
Nästan värt att behålla denna öppen som en liten påminnelse. Chansen är ju god att samma sak händer någon mer gång.

**Comment by mrinlumino** [comment]
Oj, ber så mycket om ursäkt! Förstår inte vad som hände ärligt talat. Jag hade bara lagt till två filer i min egen commit och sedan gjort en fetch & merge från kodsnack-repot innan jag committade. Måste hänt något märkligt i min egen merge. Jag borde ju sett vad som var på gång när jag lade pull-requesten men var nog lite för snabb. Lovar att hålla ögonen öppna innan jag pushar något idag.

**Comment by Masssssy** [comment]
Min tanke var ju att det skulle bli smidigare med en submodul men det verkar vara mer meck i praktiken trots all den fina teorin :D

**Comment by bjoreman** [comment]
Samtidigt tycker jag det funkar rätt bra med tanke på hur många som använder submoduler.

**Comment by mrinlumino** [comment]
Gjorde en clean checkout nu och tänker strunta i att försöka synka ner alla andras ändringar till min fork så skall jag förhoppningsvis inte orsaka mer elände framöver!

**Comment by bjoreman** [comment]
Äsch, vi lär oss ju alla saker på kuppen. Kör på!

**Comment by k0nserv** [comment]
Så här gör jag när jag uppdatera

```
git checkout master
git fetch --all
git reset --hard upstream/master
cd k0nserv-rust
git pull
cd ..
git add k0nserv-rust
git commit
git push origin master
```
